src/shortestpaths/desopo-pape.jl

```diff
@@ -0,0 +1,46 @@
+"""
+    struct DEposoPapeState{T, U}
+
+An [`AbstractPathState`](@ref) designed for D`Esopo-Pape shortest-path calculations.
+"""
+struct DEsopoPapeState{T <:Real, U <: Integer} <: AbstractPathState
+    parents::Vector{U}
+    dists::Vector{T}
+end
+
+function desopo_pape_shortest_paths(g::AbstractGraph, 
+    src::Integer,
+    distmx::AbstractMatrix{T} = weights(g)) where T <: Real
+    U = eltype(g)
+    nvg = nv(g)
+    (src in 1:nvg) || throw(DomainError("src should be in between 1 and $nvg"))
+    dists = fill(typemax(T), nvg)
+    parents = zeros(U, nvg)
+    state = Vector{Int8}()
+    state = fill(Int8(2), nvg)
+    q = U[src]
+    @inbounds dists[src] = 0
+    
+    @inbounds while !isempty(q)
+        @inbounds u = popfirst!(q)
```

Comment:
Putting `@inbounds` in front of the while-block, disables the boundchecks for all array access inside the block (as long as they don't happen inside a function called there, in which case, there are special rules), so you don't have to write `@inbounds` inside the loop.

Reply:
Done!